examples/translator/common/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub mod proto;
+
+use log::{info, warn};
+use proto::translator::{TranslateRequest, TranslateResponse};
+
+/// Client for a translator Node.
+pub struct TranslatorClient {
+    channel: oak::WriteHandle,
+}
+
+impl TranslatorClient {
+    /// Create a new translator Node and connection to it.
+    pub fn new() -> Option<TranslatorClient> {
+        let (wh, rh) = oak::channel_create().expect("failed to create channel");
+        let status = oak::node_create("translator", "oak_main", rh);
+        let _ = oak::channel_close(rh.handle);
+        match status {
+            Ok(_) => {
+                info!("translator client created");
+                Some(TranslatorClient { channel: wh })
+            }
+            Err(status) => {
+                warn!("failed to create translator: {:?}", status);
+                None
+            }
+        }
+    }
+    pub fn translate(&self, text: &str, from_lang: &str, to_lang: &str) -> Option<String> {
+        info!(
+            "attempt to translate '{}' from {} to {}",
+            text, from_lang, to_lang
+        );
+        let mut req = TranslateRequest::new();
+        req.text = text.to_string();
+        req.from_lang = from_lang.to_string();
+        req.to_lang = to_lang.to_string();
+
+        let rsp: oak::grpc::Result<TranslateResponse> = oak::grpc::invoke_grpc_method(
+            "/oak.examples.translator.Translator/Translate",
+            req,
+            self.channel,
+        );
+        match rsp {
+            Ok(rsp) => {
+                info!("translation '{}'", rsp.translated_text);
+                Some(rsp.translated_text)
+            }
+            Err(status) => {
+                warn!(
+                    "gRPC invocation failed: code {} msg {}",
+                    status.code, status.message
+                );
+                None
+            }
+        }
+    }
```

Comment:
I guess this code could / should also be generated at some point, perhaps add a TODO + issue? And also I would refactor it so that the auto-generatable (?) code is in a single method, while packing / unpacking protobuf from args and to return values is separate (since that will stay around anyways).